server/controllers/trialbalance.js

```diff
@@ -187,7 +187,7 @@ function checkDebtorCreditorExists(transactions) {
     // if nothing is returned, skip error report
     if (!rows.length) { return; }
 
-    // returns a promise error report
+    // returns a error report
     return createErrorReport('ERROR.ERR_MISSING_DEBTOR_CREDITOR', true, rows);
   });
 }
```

Comment:
```
function checkDocumentNumberExists(transactions) {
   var sql =
     'SELECT COUNT(pj.uuid) AS count, pj.doc_num, pj.trans_id, pj.deb_cred_uuid FROM posting_journal AS pj ' +
     'WHERE pj.trans_id IN (?) AND (pj.deb_cred_type = \'D\' OR pj.deb_cred_type = \'C\') ' +
     'GROUP BY pj.trans_id HAVING pj.doc_num IS NULL;';
 
   return db.exec(sql, [transactions])
   .then(function (rows) {
 
     // if nothing is returned, skip error report
     if (!rows.length) { return; }
 
-    // returns a promise error report
+    // returns a error report
     return createErrorReport('ERROR.WARN_MISSING_DOCUMENT_ID', false, rows);
   });
 }
```
You are checking the doc_num only in transaction which have the deb_cred_type, you should do it for each transaction.